boot/freeldr/freeldr/arch/uefi/uefidisk.c

```diff
@@ -0,0 +1,591 @@
+
+    InternalUefiDisk[DriveNumber].NumOfPartitions = i;
+    /* Convert checksum and signature to identifier string */
+    Identifier[0] = Hex[(Checksum >> 28) & 0x0F];
+    Identifier[1] = Hex[(Checksum >> 24) & 0x0F];
+    Identifier[2] = Hex[(Checksum >> 20) & 0x0F];
+    Identifier[3] = Hex[(Checksum >> 16) & 0x0F];
+    Identifier[4] = Hex[(Checksum >> 12) & 0x0F];
+    Identifier[5] = Hex[(Checksum >> 8) & 0x0F];
+    Identifier[6] = Hex[(Checksum >> 4) & 0x0F];
+    Identifier[7] = Hex[Checksum & 0x0F];
+    Identifier[8] = '-';
+    Identifier[9] = Hex[(Signature >> 28) & 0x0F];
+    Identifier[10] = Hex[(Signature >> 24) & 0x0F];
+    Identifier[11] = Hex[(Signature >> 20) & 0x0F];
+    Identifier[12] = Hex[(Signature >> 16) & 0x0F];
+    Identifier[13] = Hex[(Signature >> 12) & 0x0F];
+    Identifier[14] = Hex[(Signature >> 8) & 0x0F];
+    Identifier[15] = Hex[(Signature >> 4) & 0x0F];
+    Identifier[16] = Hex[Signature & 0x0F];
+    Identifier[17] = '-';
+    Identifier[18] = (ValidPartitionTable ? 'A' : 'X');
+    Identifier[19] = 0;
+    TRACE("Identifier: %s\n", Identifier);
+}
+
+static
+VOID
+UefiSetupBlockDevices(VOID)
+{
+    ULONG BlockDeviceIndex;
+    ULONG SystemHandleCount;
+    EFI_STATUS Status;
+    ULONG i;
+
+    UINTN handle_size = 0;
+    PcBiosDiskCount = 0;
+    UefiBootRootIdentifier = 0;
+
+    /* 1) Setup a list of boothandles by using the LocateHandle protocol */
+    Status = GlobalSystemTable->BootServices->LocateHandle(ByProtocol, &bioGuid, NULL, &handle_size, handles);
+    handles = MmAllocateMemoryWithType((handle_size + (EFI_PAGE_SIZE * 2) * 2), LoaderFirmwareTemporary);
+    Status = GlobalSystemTable->BootServices->LocateHandle(ByProtocol, &bioGuid, NULL, &handle_size, handles);
+    SystemHandleCount = handle_size / sizeof(EFI_HANDLE);
+    InternalUefiDisk = MmAllocateMemoryWithType(sizeof(INTERNAL_UEFI_DISK) * SystemHandleCount, LoaderFirmwareTemporary);
+
+    BlockDeviceIndex = 0;
+    for (i = 0; i < SystemHandleCount; ++i)
+    {
+        Status = GlobalSystemTable->BootServices->HandleProtocol(handles[i], &bioGuid, (void **) &bio);
+        if (handles[i] == PublicBootHandle)
+        {
+            OffsetToBoot = i; /* The Drive offset in a list of handles */
+        }
+		
+        if (EFI_ERROR(Status) || bio == NULL || bio->Media->BlockSize==0 || bio->Media->BlockSize > 2048)
+        {
+            TRACE("UefiSetupBlockDevices: UEFI Has found a block device thats failed, skipping\n");
+            continue;
+        }
+        if (bio->Media->LogicalPartition == FALSE)
+        {
+            TRACE("HardDisk\n");
+            PcBiosDiskCount++;
+            InternalUefiDisk[BlockDeviceIndex].ArcDriveNumber = BlockDeviceIndex;
+            InternalUefiDisk[BlockDeviceIndex].UefiRootNumber = i;
+            GetHarddiskInformation(BlockDeviceIndex + FIRST_BIOS_DISK);
+            BlockDeviceIndex++;
+        }
+        else if (handles[i] == PublicBootHandle)
+        {
```

Comment:
Missing space after "if"

Reply:
Looks like it's fixed, please confirm @julenuri .